src/storage_pool.h

```diff
@@ -16,18 +16,19 @@ struct StoragePoolCleanupHandler {
   }
 };
 
-class StoragePool : public NLVObject<virStoragePoolPtr, StoragePoolCleanupHandler>
+class StoragePool : public NLVObject<StoragePool, virStoragePoolPtr, StoragePoolCleanupHandler>
 {
 public:
   static void Initialize(Handle<Object> exports);
-  static Local<Object> NewInstance(virStoragePoolPtr handle);
+  NLV_OBJECT_STATIC_HELPERS(StoragePool);
 
 private:
   explicit StoragePool(virStoragePoolPtr handle);
   static Nan::Persistent<FunctionTemplate> constructor_template;
   static Nan::Persistent<Function> constructor;
   friend class StorageVolume;
   friend class Hypervisor;
+  friend class NLVObject;
```

Comment:
Thanks. btw. Once everything uses new helper functions, the other "friends" can be safely removed.